src/js/components/PostTweet.js

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import { postTweet } from '../services/timelineService';
+
+export default class PostTweet extends React.Component{
+    constructor(props){
+        super(props);
+        this.state = {
+            message: '',
+            error: null,
+            isLoaded: false,
+            charCount: 0
+        };
+    }
+
+    handleInputChange(event){
+        this.setState({
+            message: event.target.value,
+            charCount: event.target.value.length
+        });
+    }
+
+    postTweet(){
+        this.setState({
+            error: null,
+            isLoaded: false
+        });
+        postTweet(this.state.message).then(
+            (response) => {
+                this.setState({
+                    isLoaded: true
+                });
+            }, 
+            (error) => {
+                this.setState({
+                    error: error,
+                    isLoaded: true
+                });
+            }
+        );
+
+    }
+
+    renderReport(){
+        if(this.state.error && this.state.isLoaded){
+            return (<span className='report error'>There was an issue posting your tweet.</span>);
+        } else if(this.state.isLoaded){
+            return(<span className='report success'>Successfully posted your tweet.</span>);
+        }
+    }
+
+    render(){
+        return (
+            <div className='post-tweet-container'>
+                <div className='post-text-area'>
+                    <textarea
+                        className='post-input'
+                        value={this.state.message}
+                        onChange={(event) => this.handleInputChange(event)}
+                        maxLength="280"
+                    />
+                    <div className='char-count' data-count={this.state.charCount}>
+                        {this.state.charCount}/280
+                    </div>
+                </div>
+                <div className='post-controls'>
+                    {
+                        this.renderReport()
+                    }
+                    <button
+                        className='post-button'
+                        disabled={!this.state.message?.length || this.state.message == ''}
+                        onClick={() => this.postTweet()}
```

Comment:
you can replace `() => this.postTweet()` with `this.postTweet` method references work well in react event handlers. especially if you're not passing an event to your function

Reply:
this actually would pass the wrong 'this', which causes an error when trying to set the state

Reply:
There's to ways to control for this. 1) you can turn postTweet into an arrow function 2) you can bind `this`
This is a nitpick though